rtCommon/bidsArchive.py

```diff
@@ -0,0 +1,571 @@
+        metadataPath = incremental.metadataFilePath()
+
+        # 2) Verify we have a valid way to append the image to the archive.
+        # 3 cases:
+        # 2.0) Archive is empty and must be created
+        # 2.1) Image already exists within archive, append this Nifti to it
+        # 2.2) Image doesn't exist in archive, but rest of the path is valid for
+        # the archive; create new Nifti file within the archive
+        # 2.3) Neither image nor path is valid for provided archive; fail append
+        if self.isEmpty() and makePath:
+            incremental.writeToArchive(self.rootPath)
+            self.openDataset(self.rootPath)
+
+        elif self.pathExists(imgPath):
+            logger.debug("Image exists in archive, appending")
+            archiveImg = self.getImage(imgPath)
+
+            # Validate header match
+            if not self._verifyNiftiHeadersMatch(incremental.image,
+                                                 archiveImg):
+                raise ValidationError("Nifti headers failed validation!")
+            if not self._verifySidecarMetadataMatch(incremental.imgMetadata,
+                                                    self.getMetadata(imgPath)):
+                raise ValidationError("Image metadata failed validation!")
+
+            # Build 4-D NIfTI if archive has 3-D, concat to 4-D otherwise
+            incrementalData = getNiftiData(incremental.image)
+            archiveData = getNiftiData(archiveImg)
+
+            # If archive has something that isn't 3-D or 4-D, not sure what to
+            # do with it
+            # TODO(spolcyn): Handle this case more gracefully, or provide
+            # additional documentation that precludes this from happening
+            assert len(archiveData.shape) == 3 or len(archiveData.shape) == 4
+
+            if len(archiveData.shape) == 3:
+                archiveData = np.expand_dims(archiveData, 3)
+
+            newArchiveData = np.concatenate((archiveData, incrementalData),
+                                            axis=3)
+
+            newImg = nib.Nifti1Image(newArchiveData,
+                                     archiveImg.affine,
+                                     header=archiveImg.header)
+            newImg.update_header()
+            self.addImage(newImg, imgPath)
+
+        # 2.2) Image doesn't exist in archive, but rest of the path is valid for
+        # the archive; create new Nifti file within the archive
+        elif self.pathExists(dataDirPath) or makePath is True:
+            logger.debug("Image doesn't exist in archive, creating")
+
+            self.addImage(incremental.image, imgPath)
+            self.addMetadata(incremental.imgMetadata, metadataPath)
+
+        else:
+            raise ValidationError("No valid archive path for image and no override \
+                                   specified, can't append")
+
+    @failIfEmpty
+    def stripIncremental(self, subject: str, session: str, task: str,
```

Comment:
The function name may imply that the incremental is being removed from the BidsArchive. Maybe just getIncremental?